app/models/Address.scala

```diff
@@ -16,55 +16,135 @@
+    town
+  ) ++ List.from(postCode)
+
+  val asUKAddressTuple: (String, Option[String], Option[String], String, Option[String]) =
+    (addressLine1, addressLine2, addressLine3, town, postCode)
+
+  val asInternationalAddressTuple: (String, Option[String], Option[String], String, Option[String], String) =
+    (addressLine1, addressLine2, addressLine3, town, postCode, country)
+
+  val isManualAddress: Boolean = addressType match {
+    case ManualAddress => true
+    case LookupAddress => false
+  }
+}
 
 object Address {
 
-  private val addressReadsBuilder
-    : FunctionalBuilder[Reads]#CanBuild6[String, String, Option[String], Option[String], Option[String], String] =
-    (JsPath \ "addressLine1")
+  implicit val lookupFormat: OFormat[LookupAddress.type] = Json.format[LookupAddress.type]
+  implicit val manualFormat: OFormat[ManualAddress.type] = Json.format[ManualAddress.type]
+  implicit val addressTypeFormat: OFormat[AddressType] = Json.format[AddressType]
+
+  private val addressReadsBuilder =
+    (JsPath \ "id")
       .read[String]
-      .and((JsPath \ "addressLine2").read[String])
+      .and((JsPath \ "addressLine1").read[String])
+      .and((JsPath \ "addressLine2").readNullable[String])
       .and((JsPath \ "addressLine3").readNullable[String])
-      .and((JsPath \ "town").readNullable[String])
+      .and((JsPath \ "town").read[String])
       .and((JsPath \ "postCode").readNullable[String])
       .and((JsPath \ "countryCode").read[String])
+      .and((JsPath \ "addressType").read[AddressType])
+
   implicit val addressReads: Reads[Address] =
     addressReadsBuilder.apply(
-      (addressLine1, addressLine2, addressLine3, town, postCode, countryCode) => {
+      (id, addressLine1, addressLine2, addressLine3, town, postCode, countryCode, addressType) => {
         Address(
+          id,
           addressLine1,
           addressLine2,
           addressLine3,
           town,
           postCode,
           Country.getCountry(countryCode).getOrElse(""),
-          countryCode
+          countryCode,
+          addressType
         )
       }
     )
+
   implicit val addressWrites: Writes[Address] = Json.writes[Address]
+
+  def fromManualUKAddress(tup: (String, Option[String], Option[String], String, Option[String])): Address = {
+    val (addressLine1, addressLine2, addressLine3, town, postCode) = tup
+    Address(
+      "manual",
+      addressLine1,
+      addressLine2,
+      addressLine3,
+      town,
+      postCode,
+      "United Kingdom",
+      "GB",
+      ManualAddress
+    )
+  }
+
+  def fromManualInternationalAddress(
+    tup: (String, Option[String], Option[String], String, Option[String], String)
+  ): Address = {
+    val (addressLine1, addressLine2, addressLine3, town, postCode, country) = tup
+    Address(
+      "manual",
+      addressLine1,
+      addressLine2,
+      addressLine3,
+      town,
+      postCode,
+      country,
+      "N/A", // TODO get country code from country
```

Comment:
Is this for later on? If so do we have a ticket for it?

Reply:
I remember Michael, Josh and I had a meeting around this - pretty sure Micheal created a ticket but just asked